configs/configupgrade/analysis.go

```diff
@@ -177,7 +177,7 @@ func (u *Upgrader) analyze(ms ModuleSources) (*analysis, error) {
 
 				var fqn addrs.Provider
 				if providerKey == "" {
-					fqn = rAddr.DefaultProvider()
+					fqn = addrs.NewLegacyProvider(rAddr.ImpliedProvider())
```

Comment:
Since you've moved that LegacyProvider change work here, should there be a comment about what this is doing/why? I'm guessing it doesn't need a TODO/bc this is configupgrade

Reply:
Good point, I'm hoping to just remove this code before 013 but that's no reason to skip the comment.